### DISCUSSION TITLE
Lookup

### DISCUSSION THREAD

**Discussion by fastlink30** [dotnetcore/BootstrapBlazor#2567] [Q&A] [answered]
Hi,
There are demo or code to show how(and if) lookup in table is bind to list? from the web demo is difficult understand how lookup works

**Comment by ArgoZhang** [comment]
@fastlink30 sure. could you provider a demo project? i will submit PR for you

**Comment by fastlink30** [comment]
I've used template and modified page TableDemo, hope can understand what my intention was, there are 2 grid,first with TestClass, 2nd with SelectedItem, i want show text associated with Value on column Value
[BootstrapBlazorApp.OnlyServer1.zip](https://github.com/dotnetcore/BootstrapBlazor/files/13660009/BootstrapBlazorApp.OnlyServer1.zip)

**Comment by ArgoZhang** [comment, marked as answer]
@fastlink30 follow the steps

### Inject service

Program.cs
```
builder.Services.AddSingleton(typeof(ILookupService), typeof(FooLookupService));
```

### Implement FooLookupService

\\Services\\FooLookupService.cs

```
using BootstrapBlazor.Components;

namespace BootstrapBlazorApp.OnlyServer1.Services;

class FooLookupService : ILookupService
{
    public IEnumerable<SelectedItem>? GetItemsByKey(string? key)
    {
        IEnumerable<SelectedItem>? ret = null;

        if (key == "TestClass.Value")
        {
            ret = new SelectedItem[]
            {
                new("6a3413db-a958-47d0-877e-6b698fc0949f", "LookupService-Test-1"),
                new("07b2ae67-8159-497d-b6f3-1145cd9a238b", "LookupService-Test-2"),
                new("29ec481a-b50e-420a-8166-52993817fdd2", "LookupService-Test-3")
            };
        }
        return ret;
    }
}
```

### Use LookupService

TableDemo.razor
```
<Table TItem="TestClass" IsPagination="true" class="table-demo"
       Items="@testData1"
       IsStriped="true" IsBordered="true" ShowSkeleton="true" IsMultipleSelect="true"
       ShowToolbar="true" ShowSearch="true" ShowExtendButtons="true"
       AutoGenerateColumns="true" EditMode="EditMode.Popup">
    <TableColumns>
        <TableColumn @bind-Field="@context.Text1" />
        <TableColumn @bind-Field="@context.Text2" />
        <TableColumn @bind-Field="@context.Value" LookupServiceKey="TestClass.Value" />
    </TableColumns>
</Table>
```

The key point `LookupServiceKey="TestClass.Value"` then match the key in Lookup service  `key == "TestClass.Value"`

**Comment by ArgoZhang** [comment]
update project zip package [BootstrapBlazorApp.OnlyServer1.zip](https://github.com/dotnetcore/BootstrapBlazor/files/13669018/BootstrapBlazorApp.OnlyServer1.zip)

**Comment by fastlink30** [comment]
Thanks for the reply,
the solution doesn't satisfy me completely, if in my project I need to connect 10 lookups? do I have to inject 10 classes?? it's not the best, furthermore in my project I have already loaded the data for the lookups, I don't want to load them inside the injected class, I think it would be better to have a different mechanism for managing lookups within the table, perhaps being able to specify a List<SelectedItem> to be used in the tablecolumn for lookups

**Comment by ArgoZhang** [comment]
@fastlink30 only one inject class

add `if` `else if` check the `LookupServiceKey`

**Comment by ArgoZhang** [comment]
@fastlink30 the another way use `Lookup` instead `LookupServiceKey`

**Comment by fastlink30** [comment]
Lookup parameter accept only SelectedItem class?
